checkstyle-checks/src/main/resources/sonatype/checkstyle-configuration.xml

```diff
@@ -0,0 +1,181 @@
+<!--
+
+    Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
+    Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
+    "Sonatype" is a trademark of Sonatype, Inc.
+
+-->
+<!DOCTYPE module PUBLIC
+    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+<module name="Checker">
+  <property name="charset" value="UTF-8"/>
+
+  <property name="severity" value="error"/>
+
+  <property name="fileExtensions" value="java"/>
+  <module name="FileTabCharacter">
+    <property name="eachLine" value="true"/>
+  </module>
+
+  <module name="NewlineAtEndOfFile"/>
+
+  <module name="SuppressWarningsFilter" />
+
+  <module name="TreeWalker">
+    <module name="com.sonatype.checks.ClassStructureEmptyLineCheck" />
+    <module name="SuppressWarningsHolder" />
+    <module name="UnnecessaryParentheses"/>
+    <module name="EmptyLineSeparator">
+      <property name="allowMultipleEmptyLines" value="false" />
+      <property name="allowMultipleEmptyLinesInsideClassMembers" value="false" />
+      <property name="tokens" value="IMPORT, CLASS_DEF, INTERFACE_DEF, ENUM_DEF, STATIC_INIT, INSTANCE_INIT, METHOD_DEF, CTOR_DEF, VARIABLE_DEF"/>
+    </module>
+    <module name="OuterTypeFilename"/>
+    <module name="LineLength">
+      <property name="max" value="120"/>
```

Comment:
I recently came across a method which due to its return type (`Map<LongTypeName, LongTypeName>`) and verbose name already passed beyond the limit and my Eclipse formatter didn't want to wrap any other way. I don't want to spend my time of having to post-fix the IDE formatter, so be sure to verify the formatters always yield compliant code regarding the line length or this needs to be a warning and nothing that would fail the build.

Reply:
> so be sure to verify the formatters always yield compliant code regarding the line length or this needs to be a warning and nothing that would fail the build.

Yes, that is what I'm working on today.  I'll have another pull request with any formatter changes.

Reply:
As long as the delivery order matches: First the tooling to comply, then the enforcement.

Reply:
> As long as the delivery order matches: First the tooling to comply, then the enforcement.

Understood.  That is my intent.